bundles/io/org.eclipse.smarthome.io.json/src/main/java/org/eclipse/smarthome/io/json/internal/gson/OsgiGsonBindingService.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.json.internal.gson;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.io.json.JsonBindingService;
+import org.eclipse.smarthome.io.json.gson.GsonBindingService;
+import org.eclipse.smarthome.io.json.gson.GsonTypeAdapterProvider;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * JSON data binding implementation that uses Google Gson to serialize and deserialize between
+ * Java objects and JSON.
+ *
+ * @author Flavio Costa - Initial contribution and API
+ *
+ * @param <T> Type to be serialized or deserialized.
+ */
+@Component(immediate = true, service = JsonBindingService.class, property = {
+        OsgiGsonBindingService.FORMAT_OUTPUT_PROPERTY + ":Boolean=false" })
+public class OsgiGsonBindingService<T> extends GsonBindingService<T> {
+
+    /**
+     * Logger for this class.
+     */
+    private final Logger logger = LoggerFactory.getLogger(OsgiGsonBindingService.class);
+
+    /**
+     * Component property that determines whether JSON output should be formatted (easier to read) or not (less data to
+     * be transmitted).
+     */
+    static final String FORMAT_OUTPUT_PROPERTY = "formatted-output";
```

Comment:
private?

Reply:
I wanted to make it `private`, but the constant is used in the `@Component` annotation and the class does not compile if the field is made `private`.